src/utils/useUniqueId.js

```diff
@@ -1,7 +1,8 @@
 import shortid from './shortId';
 import { useState } from 'react';
 
-export default () => {
+export default (defaultId) => {
+    if (defaultId) return defaultId;
     const [uniqueId, setUniqueId] = useState();
```

Comment:
I think this would qualify as conditionally calling a hook, breaking the rules of hooks.
If the first call from an initial render of a component didn't have a defaultId, it would call `useState`, if the next call did have defaultId it wouldn't call `useState`